src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let clonedState = { ...state };
+
+  for (const action in actions) {
+    const point = actions[action];
+
+    if (point.type === 'addProperties') {
+      const resultAction = {
+        ...clonedState, ...point.extraData,
+      };
+
+      result.push(resultAction);
```

Comment:
`result.push(resultAction)` repeated each time, so you can move it from `if` blocks and push just `clonedState`